code/modules/atmospherics/machinery/components/fusion/hfr_procs.dm

```diff
@@ -318,10 +318,106 @@
-	radiation_pulse(loc, power_level * 7000, (1 / (power_level + 5)), TRUE)
-	empulse(loc, power_level * 5, power_level * 7, TRUE)
-	var/list/around_turfs = circlerangeturfs(src, power_level * 5)
+	var/flash_explosion = 0
+	var/light_impact_explosion = 0
+	var/heavy_impact_explosion = 0
+	var/devastating_explosion = 0
+	var/em_pulse = selected_fuel.meltdown_flags & HYPERTORUS_FLAG_EMP
+	var/rad_pulse = selected_fuel.meltdown_flags & HYPERTORUS_FLAG_RADIATION_PULSE
+	var/emp_light_size = 0
+	var/emp_heavy_size = 0
+	var/rad_pulse_size = 0
+	var/rad_pulse_strength = 0
+	var/gas_spread = 0
+	var/gas_pockets = 0
+	var/critical = FALSE
+
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_BASE_EXPLOSION)
+		flash_explosion = power_level * 3
+		light_impact_explosion = power_level * 2
+
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_MEDIUM_EXPLOSION)
+		flash_explosion = power_level * 6
+		light_impact_explosion = power_level * 5
+		heavy_impact_explosion = power_level * 0.5
+
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_DEVASTATING_EXPLOSION)
+		flash_explosion = power_level * 8
+		light_impact_explosion = power_level * 7
+		heavy_impact_explosion = power_level * 2
+		devastating_explosion = power_level
+
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_MINIMUM_SPREAD)
+		if(em_pulse)
+			emp_light_size = power_level * 3
+			emp_heavy_size = power_level * 1
+		if(rad_pulse)
+			rad_pulse_size = (1 / (power_level + 1))
+			rad_pulse_strength = power_level * 3000
+		gas_pockets = 5
+		gas_spread = power_level * 2
+
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_MEDIUM_SPREAD)
+		if(em_pulse)
+			emp_light_size = power_level * 5
+			emp_heavy_size = power_level * 3
+		if(rad_pulse)
+			rad_pulse_size = (1 / (power_level + 3))
+			rad_pulse_strength = power_level * 5000
+		gas_pockets = 7
+		gas_spread = power_level * 4
+
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_BIG_SPREAD)
+		if(em_pulse)
+			emp_light_size = power_level * 7
+			emp_heavy_size = power_level * 5
+		if(rad_pulse)
+			rad_pulse_size = (1 / (power_level + 5))
+			rad_pulse_strength = power_level * 7000
+		gas_pockets = 10
+		gas_spread = power_level * 6
+
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_MASSIVE_SPREAD)
+		if(em_pulse)
+			emp_light_size = power_level * 9
+			emp_heavy_size = power_level * 7
+		if(rad_pulse)
+			rad_pulse_size = (1 / (power_level + 7))
+			rad_pulse_strength = power_level * 9000
+		gas_pockets = 15
+		gas_spread = power_level * 8
+
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_CRITICAL_MELTDOWN)
+		critical = TRUE
+		Cinematic(CINEMATIC_SELFDESTRUCT,world)
+
+	explosion(
+		origin = src,
+		devastation_range = critical ? devastating_explosion * 2 : devastating_explosion,
+		heavy_impact_range = critical ?  heavy_impact_explosion * 2 : heavy_impact_explosion,
+		light_impact_range = light_impact_explosion,
+		flash_range = flash_explosion,
+		adminlog = TRUE,
+		ignorecap = TRUE
```

Comment:
How high does this explosion cap out at. Since you're ignoring the max cap and all

Reply:
with hypernob and trit/h2 the maxcap of the light impact is around the size of the SM delamination, the hypernob+antinob one is bigger but the others are a lot smaller

Reply:
How big is bigger? do you have like, a rough estimate?

Reply:
it should be a bit more than half time bigger, turf more turf less

Reply:
the max explosion size is 
devastation range 12
heavy range 24
light range 42
this is for an antinob+hypernob SM

Reply:
SM?

Reply:
derp HFR